### PR TITLE
Skip saving checkpoints during training

### PR DESCRIPTION
**Pull request by abhibyreddi**
Currently checkpoints are saved during `trainer.fit` which is not needed because we manually save checkpoints by calling `trainer.save_checkpoint`

- [X] Tests pass - ran locally on a VM with 4 GPUs and on a GKE cluster with 2 nodes
- [X] Appropriate changes to documentation are included in the PR - N/A

### CODE REVIEW

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py [hunk 1]
```diff
@@ -72,29 +69,17 @@ def main(project: str,
...
-    checkpoint_callback = ModelCheckpoint(
-        save_top_k=1 if save_only_latest else -1,
-        every_n_train_steps=1,
-        filename="checkpoint-{epoch:02d}-{step:02d}",
-        enable_version_counter=True,
-    )
-
     strategy = get_strategy(args.strategy, model, ckpt_dir_path)
-    min_epochs_save = int(os.environ.get("MIN_EPOCHS_SAVE", 4))
-    max_epochs_save = int(os.environ.get("MAX_EPOCHS_SAVE", 5))
```
Comment (MattIrv) [inline dataflux_pytorch/benchmark/checkpointing/multinode/train.py:87]: Can you update the readme to remove these as well? And possibly the deployment yaml file too if they're set there

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py [hunk 2]
```diff
@@ -59,10 +59,7 @@ def get_strategy(choice, model, ckpt_dir_path):
     return strategy
 
 
-def main(project: str,
```
Comment (MattIrv) [inline dataflux_pytorch/benchmark/checkpointing/multinode/train.py:62]: nit: As much as possible it's probably better to try to leave os.environ calls inside of main or the `if __name__ == '__main__'` block, and pass parameters to functions like `get_strategy`. It's not a major difference, but it ensures they can be cleanly reused outside of this code if needed

**Review by MattIrv — APPROVED** [review]

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py (hunk 1 above)
Comment (abhibyreddi) [reply dataflux_pytorch/benchmark/checkpointing/multinode/train.py:87]: These vars are not set in the deployment file. I renamed `MAX_STEPS_SAVE` and `MAX_STEPS_RESTORE` to `NUM_SAVE_CALLS` and `NUM_LOAD_CALLS` and updated the README. Do we really need to have these variables configurable?

**Review by awonak — APPROVED** [review]
Nice improvements!

I realized that we didn't add a note in the README for the different strategy command line args like `dataflux_fsdp` and `fsspec_fsdp`. Can you add those details here or in a follow up PR?

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py (hunk 2 above)
Comment (Yash9060) [reply dataflux_pytorch/benchmark/checkpointing/multinode/train.py:62]: +1

**Review by Yash9060 — APPROVED** [review]

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py (hunk 2 above)
Comment (abhibyreddi) [reply dataflux_pytorch/benchmark/checkpointing/multinode/train.py:62]: Done.

**Pull request merged**